azure-client-runtime/src/main/java/com/microsoft/azure/v2/LocationPollStrategy.java

```diff
@@ -29,34 +30,39 @@
      */
     public static final String HEADER_NAME = "Location";
 
-    private LocationPollStrategy(RestProxy restProxy, String fullyQualifiedMethodName, String locationUrl, long delayInMilliseconds) {
-        super(restProxy, delayInMilliseconds);
+    private LocationPollStrategy(RestProxy restProxy, SwaggerMethodParser methodParser, String locationUrl, long delayInMilliseconds) {
+        super(restProxy, methodParser, delayInMilliseconds);
 
-        this.fullyQualifiedMethodName = fullyQualifiedMethodName;
         this.locationUrl = locationUrl;
     }
 
     @Override
     public HttpRequest createPollRequest() {
-        return new HttpRequest(fullyQualifiedMethodName, "GET", locationUrl);
+        return new HttpRequest(fullyQualifiedMethodName(), "GET", locationUrl);
     }
 
     @Override
     public Single<HttpResponse> updateFromAsync(HttpResponse httpPollResponse) {
-        final int httpStatusCode = httpPollResponse.statusCode();
-
-        updateDelayInMillisecondsFrom(httpPollResponse);
-
-        if (httpStatusCode == 202) {
-            String newLocationUrl = httpPollResponse.headerValue(HEADER_NAME);
-            if (newLocationUrl != null) {
-                locationUrl = newLocationUrl;
-            }
-        }
-        else {
-            done = true;
-        }
-        return Single.just(httpPollResponse);
+        return ensureExpectedStatus(httpPollResponse, new int[] {202})
```

Comment:
Small, maybe immeasurable thing: these kinds of constant arrays can be static final to reduce allocations.